src/transformers/configuration_utils.py

```diff
@@ -125,7 +125,7 @@ def num_labels(self, num_labels):
         self.label2id = dict(zip(self.id2label.values(), self.id2label.keys()))
         self.label2id = dict((key, int(value)) for key, value in self.label2id.items())
 
-    def save_pretrained(self, save_directory):
+    def save_pretrained(self, save_directory, use_diff=True):
```

Comment:
Need to document the additional argument in the docstring.

@LysandreJik we have a script testing that the code of the docstring examples run fine, right?
Could we maybe one day make a similar script that tests that all the arguments in methods signatures with docstrings are mentioned in the associated docstring? I feel like I keep asking people to document their new arguments in the docstring, would be nice to just have a test that fails if it's not done (cc @julien-c @sshleifer)

(I can't think of a good reason you wouldn't want to document an argument in the docstring)

Reply:
Indeed, usually the IDE does that, but we could definitely look into enforcing it. I can't think of a reason we would want for an argument to not be documented either.

Reply:
I think https://pypi.org/project/flake8-docstrings/ might be what we're looking for, haven't used it however.

Reply:
I feel like this should be the only option (i.e., remove the argument completely). i.e., is there really a case where we want the full JSON serialization of the object?

Or at least, remove it from `save_pretrained`, keep it only on `to_json_string()` (and then it can default to True there, maybe it's more correct)

Reply:
(the user of save_pretrained does not really need to know which exact serialization method we use underneath, IMO)

Reply:
I don't know either but if we can keep a backward compatible option with this flag, why not maybe?

It's kinda hard to think about all the use cases people may have (e.g. loading in an older `pytorch-transformers` version the configuration which might break now for them).